flink-python/pyflink/common/execution_config.py

```diff
@@ -673,7 +674,7 @@ def is_use_snapshot_compression(self):
         """
         return self._j_execution_config.isUseSnapshotCompression()
 
-    def set_use_snapshot_compression(self, use_snapshot_compression):
+    def set_use_snapshot_compression(self, use_snapshot_compression: bool):
         """
```

Comment:
What about change ExecutionMode, InputDependencyConstraint, etc to enum type, then we could also add type hints for set_execution_mode/get_execution_mode, get_default_input_dependency_constraint/set_default_input_dependency_constraint?

Reply:
Yes. Make sense.